cli/src/merge_tools/external.rs

```diff
@@ -286,8 +289,49 @@ pub fn run_mergetool_external(
         }),
         Err(new_file_ids) => conflict.with_new_file_ids(&new_file_ids),
     };
-    let mut tree_builder = MergedTreeBuilder::new(tree.id());
     tree_builder.set_or_remove(repo_path.to_owned(), new_tree_value);
+    Ok(())
+}
+
+pub fn run_mergetool_external(
+    ui: &Ui,
+    path_converter: &RepoPathUiConverter,
+    editor: &ExternalMergeTool,
+    tree: &MergedTree,
+    merge_tool_files: &[MergeToolFile],
+    default_conflict_marker_style: ConflictMarkerStyle,
+) -> Result<MergedTreeId, ConflictResolveError> {
+    let mut tree_builder = MergedTreeBuilder::new(tree.id());
+    for (i, merge_tool_file) in merge_tool_files.iter().enumerate() {
+        writeln!(
+            ui.status(),
+            "Resolving conflicts in: {}",
+            path_converter.format_file_path(&merge_tool_file.repo_path)
+        )?;
+        match run_mergetool_external_single_file(
+            editor,
+            tree.store(),
+            merge_tool_file,
+            default_conflict_marker_style,
+            &mut tree_builder,
+        ) {
+            Ok(()) => {}
+            Err(err) if i == 0 => {
+                // Since no conflicts were successfully resolved, return the error
+                return Err(err);
+            }
+            Err(err) => {
+                // Since some conflicts were already successfully resolved, just print a warning
+                // and stop resolving conflicts
+                writeln!(
+                    ui.warning_default(),
+                    "Stopping due to error after resolving {i} conflicts"
+                )?;
+                print_error_sources(ui, Some(&err))?;
+                break;
```

Comment:
Yeah, I was considering that as well. I mainly chose to make it a warning instead of an error because I didn't want any previous successful resolutions to be discarded (so returning an error immediately isn't an option), and I thought it looked strange having the error message be at the end of the command after the transaction commits successfully instead of being before the transaction commits. Maybe I could make it so that it prints the error immediately, but doesn't actually exit with the error code until after the transaction commits?

Reply:
> Maybe I could make it so that it prints the error immediately, but doesn't actually exit with the error code until after the transaction commits?

SGTM, thanks.